src/ir/context.rs

```diff
@@ -799,6 +799,23 @@ impl BindgenContext {
         }
     }
 
+    /// Assign a new generated name for each anonymous field.
+    fn deanonymize_fields(&mut self) {
+        for item in self.items.values_mut() {
+            let ty = match item.kind_mut().as_type_mut() {
+                Some(ty) => ty,
+                None => continue,
+            };
+
+            let comp_info = match *ty.kind_mut() {
+                TypeKind::Comp(ref mut comp_info) => comp_info,
+                _ => continue,
```

Comment:
This whole dance is a little more succinct as:

```rust
item.kind_mut()
    .as_type_mut()
    .and_then(|ty| ty.as_comp_mut())
    .map(|comp| comp.deanonymize_fields());
```

Reply:
Is it really OK to have side effects inside a `map`?

Reply:
Anyway, iterators are lazy and should be consumed to provide side-effects. I see only two appropriate ways to consume Iterator here:

- `collect::<Vec<()>>`,
- `for_each`, which seems to be nightly

So I decided to do it a little bit differently and I hope you wouldn't mind!

Reply:
Note that the `map` in my snippet was `Option::map`, not `Iterator::map`, and so it is not lazy. That approach works as well!

Reply:
Ah, yeah! See it now, thanks.